mods/sbz_bio/habitat.lua

```diff
@@ -46,28 +46,38 @@ function sbz_api.habitat_tick(start_pos, meta)
     local co2 = 0
     for _, v in ipairs(habitat.co2_sources) do
         local pos, node = unpack(v)
-        co2 = co2+minetest.registered_nodes[node.name].co2_action(pos, node)
+        if stage == PcgRandom(hash(pos)):next(0, 9) then co2 = co2+minetest.registered_nodes[node.name].co2_action(pos, node) end
```

Comment:
why use pcgrandom there when math.random exists, just wondering

Reply:
previously it updated everything at once every ten seconds, but I wanted to make it so they'd all update at different times but still always be every ten seconds. I used PcgRandom so I could seed it with the position, and hence each position is updated by the habitat at the same point in each ten second cycle. dunno if that makes sense

Reply:
oh ok